comments-bundle/src/EventListener/DataContainer/CommentsConditionalFieldsListener.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CommentsBundle\EventListener\DataContainer;
+
+use Contao\CoreBundle\DataContainer\PaletteManipulator;
+use Contao\CoreBundle\DependencyInjection\Attribute\AsHook;
+use Contao\System;
+
+#[AsHook('loadDataContainer')]
+class CommentsConditionalFieldsListener
+{
+    public function __invoke(string $table): void
+    {
+        if (!isset($GLOBALS['TL_DCA'][$table])) {
+            return;
+        }
+
+        switch ($table) {
+            case 'tl_module':
+                self::applyModuleFields();
```

Comment:
What is the advantage of `self::` over `$this->`? I don't think we are using `self::` for non-static method calls anywhere else.

Reply:
Was done out of habit as I don't use the current object but rather the method in the class itself.
Either we could make the 3 methods static or refer to $this->?

Reply:
Was done out of habit due to referring to the class. Will change that.